connectors/es/client.py

```diff
@@ -221,12 +221,12 @@ async def execute_with_retry(self, func):
 
                 return result
             except ConnectionTimeout:
-                self._logger.debug(f"Attempt {retry}: connection timeout")
+                self._logger.warning(f"Attempt {retry}: connection timeout")
```

Comment:
I don't love the idea of having a bunch of WARN logs that don't tell you _what_ is being retried. Is there a way we can get the name of `func` or anything so that this can give you a little more information?

Reply:
Good call, I've added this code in https://github.com/elastic/connectors/pull/2155/commits/e6dda428e005d7457ae7b2e39b2c0b9e0fbf4e29

Reply:
Example log line would look like:

```
[FMWK][15:15:29][WARNING][Connector id: RmdNrY0Bo-hBYCnOmSCv, index name: search-outlook, Sync job id: VWdTrY0Bo-hBYCnOLyBy] Client method 'search' retry 1: api error with status 429
```

Reply:
❤️ love it!